lib/pure/oids.nim

```diff
@@ -51,45 +48,60 @@ proc parseOid*(str: cstring): Oid =
     bytes[i] = chr((hexbyte(str[2 * i]) shl 4) or hexbyte(str[2 * i + 1]))
     inc(i)
 
-proc oidToString*(oid: Oid, str: cstring) =
-  ## Converts an oid to `str` which must have space allocated for 25 elements.
+template toStringImpl[T: string | cstring](result: var T, oid: Oid) =
```

Comment:
Imho since cstring now works in the VM there is no reason for this template. Also `streams` module is using `cstring` cast in places.

Reply:
This template is needed so that oidToString can be deprecated